Main/ClientApp/src/Components/SpotifyCallbackPage.tsx

```diff
@@ -1,17 +1,47 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import UsersStore from "../Stores/UsersStore";
 import { Redirect } from "react-router-dom";
-import { UrlHash } from "../Infrastructure/UrlUtilities";
+import { GetQueryParams } from "../Infrastructure/UrlUtilities";
 import Cookies from "js-cookie";
-import { useCookie } from "../Infrastructure/CustomReactHooks";
+import { SpotifyTokenCookie } from "../Model/Spotify";
 
 export default function SpotifyCallbackPage() {
+  const [currentUser, setCurrentUser] = useState(UsersStore.getCurrentUser());
   const hash = window.location.hash;
-  let params = UrlHash(hash);
+  let params = GetQueryParams(hash);
 
-  Cookies.set("spotifyUserToken", JSON.stringify(params.get("access_token")));
-  let [collectionId, _] = useCookie<number | null>(
-    "spotifyRediretCollectionId",
-    null
+  useEffect(() => {
+    const handler = () => {
+      setCurrentUser(UsersStore.getCurrentUser());
+    };
+    UsersStore.addChangeListener(handler);
+    return () => UsersStore.removeChangeListener(handler);
+  }, []);
+
+  let collectionId = Cookies.getJSON("spotifyRedirectCollectionId") as
+    | number
```

Comment:
does the | before number do anything here? Seems like it should not work

Reply:
VSCode does that when I save. I know that it looks weird but it somehow works